### PR TITLE
Fake /proc/1/cgroups for Xen-Test running in kubernetes

### LINKED ISSUE #61466: [BUG] test_xen_virtual fails if run in a Kubernetes container
**Description**
test_xen_virtual fails if the tests for grains are executed in a kubernetes container

**Setup**
Started an Ubuntu 20.04 container on k3s, cloned the current salt repository from Github and tried to run the tests

Please be as specific as possible and give set-up details.

- [ ] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [x] container (Kubernetes via k3s, running an Ubuntu 20.04 container)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD

  
**Steps to Reproduce the behavior**
```
root@ubuntu-94456ffb5-gwzqj:~/salt# nox -e 'pytest-zeromq-3(coverage=False)' -- tests/pytests/unit/grains
nox > Running session pytest-zeromq-3(coverage=False)
nox > Re-using existing virtual environment at .nox/pytest-zeromq-3-coverage-false.
nox > Session pytest-zeromq-3(coverage=False) was successful.
nox > Running session pytest-parametrized-3(crypto=None, transport='zeromq', coverage=False)
nox > Re-using existing virtual environment at .nox/pytest-parametrized-3-crypto-none-transport-zeromq-coverage-false.
nox > python -m pip install --progress-bar=off -U 'pip>=20.2.4,<21.2' 'setuptools!=50.*,!=51.*,!=52.*,<59' wheel
nox > python -m pip install --progress-bar=off -r requirements/static/ci/py3.8/linux.txt
nox > python -m pytest --rootdir /root/salt --log-file=artifacts/logs/runtests-20220115220831.394020.log --log-file-level=debug --show-capture=no -ra -s --transport=zeromq tests/pytests/unit/grains
Removing stale tempdir: /tmp/stsuite
================================================================= test session starts =================================================================
platform linux -- Python 3.8.10, pytest-6.1.2, py-1.10.0, pluggy-0.13.0
tempdir: /tmp/stsuite
max open files; soft: 1048576; hard: 1048576
rootdir: /root/salt, configfile: pytest.ini
plugins: tempdir-2019.10.12, helpers-namespace-2021.4.29, salt-factories-0.907.0, subtests-0.4.0, flaky-3.7.0
collected 103 items

tests/pytests/unit/grains/test_core.py ............s........................s.......................................................................................F.............................s....sss.....s............

====================================================================== FAILURES =======================================================================
__________________________________________________________________ test_xen_virtual ___________________________________________________________________

    @pytest.mark.skip_unless_on_linux
    def test_xen_virtual():
        """
        Test if OS grains are parsed correctly for Xen hypervisors
        """
        with patch.multiple(
            os.path,
            isdir=MagicMock(
                side_effect=lambda x: x
                in ["/sys/bus/xen", "/sys/bus/xen/drivers/xenconsole"]
            ),
        ), patch.dict(core.__salt__, {"cmd.run": MagicMock(return_value="")}), patch.dict(
            core.__salt__,
            {"cmd.run_all": MagicMock(return_value={"retcode": 0, "stdout": ""})},
        ):
>           assert (
                core._virtual({"kernel": "Linux"}).get("virtual_subtype") == "Xen PV DomU"
            )
E           AssertionError: assert 'kubernetes' == 'Xen PV DomU'
E             - Xen PV DomU
E             + kubernetes

tests/pytests/unit/grains/test_core.py:1439: AssertionError
================================================================== warnings summary ===================================================================
salt/ext/tornado/web.py:1805
  /root/salt/salt/ext/tornado/web.py:1805: DeprecationWarning: invalid escape sequence \.
    """A collection of request handlers that make up a web application.

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=============================================================== short test summary info ===============================================================
SKIPPED [1] tests/pytests/unit/grains/test_core.py:284: Platform is not Windows, skipped
SKIPPED [1] tests/pytests/unit/grains/test_core.py:1080: Platform is not Windows, skipped
SKIPPED [1] tests/pytests/unit/grains/test_core.py:2267: Not Missing dateutil.tz
SKIPPED [1] tests/pytests/unit/grains/test_core.py:2359: Platform is not Windows, skipped
SKIPPED [1] tests/pytests/unit/grains/test_core.py:2387: Platform is not Windows, skipped
SKIPPED [1] tests/pytests/unit/grains/test_core.py:2416: Platform is not Windows, skipped
SKIPPED [1] tests/pytests/unit/grains/test_core.py:2568: Platform is not Windows, skipped
FAILED tests/pytests/unit/grains/test_core.py::test_xen_virtual - AssertionError: assert 'kubernetes' == 'Xen PV DomU'
================================================= 1 failed, 95 passed, 7 skipped, 1 warning in 2.00s ==================================================
nox > Command python -m pytest --rootdir /root/salt --log-file=artifacts/logs/runtests-20220115220831.394020.log --log-file-level=debug --show-capture=no -ra -s --transport=zeromq tests/pytests/unit/grains failed with exit code 1
nox > Session pytest-parametrized-3(crypto=None, transport='zeromq', coverage=False) failed.
nox > Ran multiple sessions:
nox > * pytest-zeromq-3(coverage=False): success
nox > * pytest-parametrized-3(crypto=None, transport='zeromq', coverage=False): failed
```

**Expected behavior**
I would not expect this test to fail.

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
root@ubuntu-94456ffb5-gwzqj:~/salt# salt --versions-report
Salt Version:
          Salt: 3004

Dependency Versions:
          cffi: Not Installed
      cherrypy: Not Installed
      dateutil: Not Installed
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 3.0.3
       libgit2: Not Installed
      M2Crypto: Not Installed
          Mako: Not Installed
       msgpack: 1.0.3
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.12.0
        pygit2: Not Installed
        Python: 3.8.10 (default, Nov 26 2021, 20:14:08)
  python-gnupg: Not Installed
        PyYAML: 6.0
         PyZMQ: 21.0.2
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.3

System Versions:
          dist: ubuntu 20.04 focal
        locale: utf-8
       machine: x86_64
       release: 5.4.0-92-generic
        system: Linux
       version: Ubuntu 20.04 focal
```
</details>

### PR DESCRIPTION
**Pull request by mayrstefan**
### What does this PR do?
The test for Xen fails if it is running in kubernetes. So we mock /proc/1/cgroups that it does not detect kubernetes instead of Xen

### What issues does this PR fix or reference?
Fixes: #61466

### Previous Behavior
Running in kubernetes the assert for xen_virtual failed because it detected kubernetes

### New Behavior
Testing for Xen detects Xen

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: changelog/61466.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+Fix test for Xen virtual_subtype if run in Kubernetes
```
Comment (Ch3LL) [inline changelog/61466.fixed:1]: Can you remove this file? Changelog files are not needed for test fixes.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: changelog/61466.fixed (hunk 1 above)
Comment (mayrstefan) [reply changelog/61466.fixed]: Sure. It looks like today the automated Jenkins tests are flaky again.

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
Those were legitimate failures not related to your PR. There is a fix in for them now so I'll update your branch and they should not be failing anymore

**Comment by dmurphy18** [comment]
@mayrstefan Can you fix the pre-commit issue, it is simple code formatting

**Review by Ch3LL — DISMISSED** [review]

**Comment by Ch3LL** [comment]
Can you fix pre-commit again. You can use the diff that the jenkins output provides to make things easier or run pre-commit on your code.

**Comment by mayrstefan** [comment]
Such a simple change and so many things I fail to do right. Sorry.
Where can I find that pre-commit script to check it locally before commiting any changes?

**Comment by Ch3LL** [comment]
No worries at all :) Just good practice for next time. 

Here is the documentation for pre-commit https://docs.saltproject.io/en/latest/topics/development/contributing.html#pre-commit-and-nox-setup

If you want to run pre-commit without running `git commit -m <message>` you can run `pre-commit run --color always --show-diff-on-failure -a -v `

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**